src/Vendr.Contrib.PaymentProviders.QuickPay/QuickPayCheckoutPaymentProvider.cs

```diff
@@ -71,10 +74,53 @@ public override PaymentFormResult GenerateForm(OrderReadOnly order, string conti
                     var clientConfig = GetQuickPayClientConfig(settings);
                     var client = new QuickPayClient(clientConfig);
 
+                    var reference = order.OrderNumber;
+
+                    // QuickPay has a limit of order id between 4-20 characters.
+                    if (reference.Length > 20)
+                    {                       
+                        var store = Vendr.Services.StoreService.GetStore(order.StoreId);
+                        var orderNumberTemplate = store.OrderNumberTemplate;
+
+                        // If the ordernumber template is manipulatet, we then remove everything other than the VENDR automated ordernumber.
```

Comment:
The order number template doesn't need to be manipulated. The default order number template `ORDER-{0}` is generating a too long order number from v1.8.6 https://vendr.net/docs/core/changelog/#v186

It basically only handle this further checks if the order number template is not equal `{0}`.

Reply:
Hi Bjarne.

That's right, but i consider it manipulated by the developer, when its anything else than {0}.